eng/scripts/Tests/Copy-ApiVersion/Copy-ApiVersion.Tests.ps1

```diff
@@ -0,0 +1,56 @@
+Import-Module Pester
+
+. "$PSScriptRoot\..\..\Copy-ApiVersion-Functions.ps1"
+
+Describe "Copy-ApiVersion" {
+    Context "Update provider Readme file" {
+        It "Should generate a new tag section with a list of all specs files for the new version" -TestCases @(
+            @{
+                version = "2024-01-01-preview"
+                provider = "Microsoft.AgFoodPlatform"
+                versionStatus = "preview"
+                specFiles = @("agfood.json")
+                expected = "newTag-Test1.md"
+                providerSubFolder = $null
+            },
+            @{
+                version = "2024-01-01"
+                provider = "Microsoft.Compute"
+                versionStatus = "preview"
+                specFiles = @("availabilitySet.json","capacityReservation.json","computeRPCommon.json","dedicatedHost.json","image.json","logAnalytic.json","proximityPlacementGroup.json","restorePoint.json","runCommand.json","sshPublicKey.json","virtualMachine.json","virtualMachineExtensionImage.json","virtualMachineImage.json","virtualMachineScaleSet.json")
+                expected = "newTag-Test2.md"
+                providerSubFolder = "ComputeRP"
+            }
+        ) {
+            param($version, $provider, $versionStatus, $specFiles, $expected, $providerSubFolder)
+            $newContentBlock = Get-NewTagSection $version $provider $versionStatus $specFiles $providerSubFolder
+            $fileToCompare = Get-Content "$PSScriptRoot\expected\$expected" -Raw
+            $newContentBlock | Should be $fileToCompare
+        }
+
+        It "Should add the new tag section and update default tag" -TestCases @(
+            @{
+                readmeContent = "input\agfood.md"
```

Comment:
Let's not copy these specs and expected instead let's run against a few under the specification folders and make sure things don't crash and do basic validation of the changes in the file to ensure it did something like you want.